# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../grub/no-menu-border.patch
# Copyright (C) 2004 - 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- g/stage2/stage2.c.orig	2005-03-19 18:51:57.000000000 +0100
+++ g/stage2/stage2.c	2005-05-23 23:59:20.000000000 +0200
@@ -97,7 +97,7 @@
 static void
 print_entry (int y, int highlight, char *entry)
 {
-  int x;
+  int x; int t = 0;
 
   if (current_term->setcolorstate)
     current_term->setcolorstate (COLOR_STATE_NORMAL);
@@ -116,10 +116,14 @@
 	  else
 	    grub_putchar (*entry++);
 	}
-      else
+      else {
 	grub_putchar (' ');
+	if (!t && current_term->setcolorstate)
+          current_term->setcolorstate (COLOR_STATE_NORMAL);
+	t = 1;
+      }
     }
-  gotoxy (74, y);
+  gotoxy (2, y);
 
   if (current_term->setcolorstate)
     current_term->setcolorstate (COLOR_STATE_STANDARD);
@@ -158,7 +162,7 @@
   else
     grub_putchar (' ');
 
-  gotoxy (74, y + entryno + 1);
+  gotoxy (2, y + entryno + 1);
 }
 
 static void
@@ -302,7 +307,8 @@
       if (current_term->flags & TERM_DUMB)
 	print_entries_raw (num_entries, first_entry, menu_entries);
       else
-	print_border (3, 12);
+	/*print_border (3, 12);*/
+	gotoxy (74, 3 + 12);
 
       grub_printf ("\n\
       Use the %c and %c keys to select which entry is highlighted.\n",
@@ -361,7 +367,7 @@
 	      gotoxy (3, 22);
 	      grub_printf ("The highlighted entry will be booted automatically in %d seconds.    ",
 			   grub_timeout);
-	      gotoxy (74, 4 + entryno);
+	      gotoxy (2, 4 + entryno);
 	  }
 	  
 	  grub_timeout--;
--- grub-0.97/stage2/stage2.c.vanilla	2008-08-20 13:14:34.000000000 +0200
+++ grub-0.97/stage2/stage2.c	2008-08-20 13:15:38.000000000 +0200
@@ -493,7 +493,7 @@
 	}
 
 menu_restart:
-
+#if 0
       /* Print the number of the current entry in the right upper corner of
        * the menu, up to 999 entries are supported, modify the coordinates
        * and putchar command to add more
@@ -548,7 +548,7 @@
 
           gotoxy(74, 4 + entryno);
         }
-
+#endif
 
       /* Check for a keypress, however if TIMEOUT has been expired
 	 (GRUB_TIMEOUT == -1) relax in GETKEY even if no key has been
